test/test-udp-open.c

```diff
@@ -164,3 +175,30 @@ TEST_IMPL(udp_open) {
   MAKE_VALGRIND_HAPPY();
   return 0;
 }
+
+
+TEST_IMPL(udp_open_twice) {
+  uv_udp_t client;
+  uv_os_sock_t sock1, sock2;
+  int r;
+
+  startup();
+  sock1 = create_udp_socket();
+  sock2 = create_udp_socket();
+
+  r = uv_udp_init(uv_default_loop(), &client);
+  ASSERT(r == 0);
+
+  r = uv_udp_open(&client, sock1);
+  ASSERT(r == 0);
```

Comment:
Suggestion: add another test asserting that calling uv_tcp_open(&client, sock1) again fails with UV_EBUSY.